src/parsers/manifestjson.js

```diff
@@ -730,6 +730,26 @@ export default class ManifestJSONParser extends JSONParser {
 
     this.validateRestrictedPermissions();
     this.validateExtensionID();
+    this.ensureHiddenAddonsWithoutBrowserActions();
+  }
+
+  ensureHiddenAddonsWithoutBrowserActions() {
```

Comment:
Nit (totally optional), maybe we can call this `validateHiddenAddon` or `validateHiddenAddonRestrictions`?